hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -262,6 +262,12 @@ protected Data removeInternal(Data key) {
     }
 
     @Override
+    protected void removeAllInternal(Predicate predicate) {
+        super.removeAllInternal(predicate);
+        nearCache.clear();
```

Comment:
You cannot return from the `remote(predicate)` call until the relevant entries are removed from near-cache. Is this true? 

Cannot the entry processor return keys it processed?